### PR TITLE
Enable more Roslynator rules

### PR DESCRIPTION
**Pull request by RoosterDragon**
Following on from #21013.

Enforces 23 Roslynator rules across the project.

I have batched together a PR for these rules as they each had limited existing violations and are hopefully uncontroversial. However fixes are arranged per commit so we can easily bikeshed over rules as desired.

Reviewing with Ignore Whitespace will make the diff in a couple of files less cumbersome.

See https://josefpihrt.github.io/docs/roslynator/analyzers for explanation of each rule.

### CODE REVIEW

**Comment by pchote** [comment]
Will this conflict with #21175?

**Comment by RoosterDragon** [comment]
> Will this conflict with #21175?

Yes.

File: OpenRA.Mods.Common/Traits/AcceptsDeliveredExperience.cs [hunk 1]
```diff
@@ -28,6 +28,5 @@ public class AcceptsDeliveredExperienceInfo : TraitInfo, Requires<GainsExperienc
 
 	public class AcceptsDeliveredExperience
 	{
-		public AcceptsDeliveredExperience() { }
 	}
```
Comment (PunkPun) [inline OpenRA.Mods.Common/Traits/AcceptsDeliveredExperience.cs]: turn it onto single line braces?

File: OpenRA.Mods.Cnc/UtilityCommands/ImportTiberianDawnMapCommand.cs [hunk 2]
```diff
@@ -34,7 +34,6 @@ public override void ValidateMapFormat(int format)
 			if (format > 1)
 			{
 				Console.WriteLine($"ERROR: Detected NewINIFormat {format}. Are you trying to import a Red Alert map?");
-				return;
 			}
```
Comment (PunkPun) [inline OpenRA.Mods.Cnc/UtilityCommands/ImportTiberianDawnMapCommand.cs]: remove braces?

File: OpenRA.Mods.Cnc/UtilityCommands/ImportRedAlertMapCommand.cs [hunk 3]
```diff
@@ -36,7 +36,6 @@ public override void ValidateMapFormat(int format)
 			if (format < 2)
 			{
 				Console.WriteLine($"ERROR: Detected NewINIFormat {format}. Are you trying to import a Tiberian Dawn map?");
-				return;
 			}
```
Comment (PunkPun) [inline OpenRA.Mods.Cnc/UtilityCommands/ImportRedAlertMapCommand.cs]: remove braces?

File: OpenRA.Game/Map/Map.cs [hunk 4]
```diff
@@ -1353,13 +1353,18 @@ public IEnumerable<CPos> FindTilesInAnnulus(CPos center, int minRange, int maxRa
...
 				{
-					var t = offset + center;
-					if (allowOutsideBounds ? Tiles.Contains(t) : Contains(t))
-						yield return t;
+					foreach (var offset in Grid.TilesByDistance[i])
+					{
+						var t = offset + center;
+						if (allowOutsideBounds ? Tiles.Contains(t) : Contains(t))
+							yield return t;
+					}
```
Comment (PunkPun) [inline OpenRA.Game/Map/Map.cs:1367]: this is quite a weird change. Why create a new IEnumerable?

**Review by PunkPun — COMMENTED** [review]
These rules seem fairly smart, overall LGTM. Left a few comments

**Comment by PunkPun** [comment]
> Will this conflict with #21175?

It doesn't look lie it conflicts, the changes that overlap are identical

File: OpenRA.Game/Map/Map.cs (hunk 4 above)
Comment (RoosterDragon) [reply OpenRA.Game/Map/Map.cs:1367]: This is [RCS1227](https://josefpihrt.github.io/docs/roslynator/analyzers/RCS1227). In the fixed form, the argument exceptions will throw immediately. The previous form doesn't trigger those exceptions until you try and enumerate the results - which might not be until much later.

**Review by PunkPun — APPROVED** [review]
LGTM

**Comment by PunkPun** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/39af2395c24ff2cbb053bfa3cf275ead819e659d...e555cab5540d75a6879a94c537e16402aa6751cf)

**Pull request merged**